exe/Main.hs

```diff
@@ -84,12 +85,15 @@ main = do
     let plugins = Completions.plugin <> CodeAction.plugin
         onInitialConfiguration = const $ Right ()
         onConfigurationChange  = const $ Right ()
+        options = def { LSP.executeCommandCommands = Just ["typesignature.add"]
+                      , LSP.completionTriggerCharacters = Just "."
+                      }
```

Comment:
What's the reason for these two changes to options? I don't see what's necessary. Is it fixing a bug? Enabling something else - and if so what?

Reply:
In haskell-language-server, the plugin mechanism allows each plugin to register a list of commands. So we only know when the plugins are assembled which ones we need to register.  Hence it needs to be an option we pass in, rather than hardcoded inside ghcide.

Similarly, we allow multiple completion providers, which may trigger on different characters depending on need.

Reply:
Yep, I can totally guess that this needs to become a list passed in and modified. But here you are just hardcoding it, without any corresponding other change anywhere, which puzzles me.

Reply:
This is the `ghcide` version.

See 
https://github.com/alanz/haskell-language-server/blob/plugin-api/exe/Main.hs#L149-L150
where I just realised I have not worked through the trigger characters yet. But they will be needed, may as well bring them in as a coherent change.